### PR TITLE
Have the frontend handle the backend going down more gracefully

### PR DESCRIPTION
**Pull request by ConorBobbleHat**
\<chanting noises\>
_enog eb snoisolpmi_
_enog eb snoisolpmi_
_enog eb snoisolpmi!_

- - - 
This PR:
- Introduces a new page for network-specific errors (and routes errors that technically aren't NetworkErrors there too, like a status-502 response from nginx). Almost every page on the site will redirect here in case of request failure.
- In case of request failure, the scratch editor now stays on-screen (albeit with stale data), and shows a banner in order to prevent losing users' work.

### CODE REVIEW

**Comment by ConorBobbleHat** [comment]
old error page:
image

new network error page:
image

new scratch editor behavior:
image

File: frontend/src/app/error.tsx [hunk 1]
```diff
@@ -7,22 +7,36 @@ import { SyncIcon } from "@primer/octicons-react"
...
+        <div className="grow" />
+        <main className="max-w-prose p-4 md:mx-auto">
+            <h1 className="py-4 text-3xl font-semibold">We're having some trouble reaching the backend</h1>
 
+            <div className="rounded bg-gray-2 p-4 text-gray-11">
+                <code className="font-mono text-sm">{error.toString()}</code>
+            </div>
+
+            <p className="py-4">
+                If your internet connection is okay, we're probably down for maintenence, and will be back shortly. If this issue persists - <a href="https://discord.gg/sutqNShRRs" className="text-blue-11 hover:underline active:translate-y-px">let us know</a>.
```
Comment (nanaian) [inline frontend/src/app/error.tsx:26]:
```suggestion
                If your internet connection is okay, we're probably down for maintenance, and will be back shortly. If this issue persists - <a href="https://discord.gg/sutqNShRRs" className="text-blue-11 hover:underline active:translate-y-px">let us know</a>.
```

**Review by nanaian — CHANGES_REQUESTED** [review]
LGTM besides typo. thanks!

File: frontend/src/app/error.tsx (hunk 1 above)
Comment (ethteck) [reply frontend/src/app/error.tsx:26]: that word forever feels incorrectly spelled

Comment (ConorBobbleHat) [reply frontend/src/app/error.tsx:26]: 🤦‍♂️

**Pull request merged**